OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -361,11 +359,16 @@ public void Activate(Player p)
 			resourceTypeIndices = new BitArray(tileset.TerrainInfo.Length); // Big enough
 			foreach (var t in Map.Rules.Actors["world"].TraitInfos<ResourceTypeInfo>())
 				resourceTypeIndices.Set(tileset.GetTerrainIndex(t.TerrainType), true);
+
+			var conditionManager = p.PlayerActor.TraitOrDefault<ConditionManager>();
+			if (conditionManager != null)
+				if (conditionToken == ConditionManager.InvalidConditionToken)
+					conditionToken = conditionManager.GrantCondition(p.PlayerActor, Info.Condition);
 		}
 
 		public void QueueOrder(Order order)
```

Comment:
Maybe add a `// DEPRECATED: Bot modules should queue orders directly` comment here?

Reply:
It's strangely not showing here nor marked as outdated, but I did.